services/httppost/service.go

```diff
@@ -0,0 +1,200 @@
+package httppost
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/influxdata/kapacitor/alert"
+	"github.com/influxdata/kapacitor/bufpool"
+)
+
+type Service struct {
+	mu        sync.RWMutex
+	endpoints map[string]Config
+	logger    *log.Logger
+}
+
+func NewService(c Configs, l *log.Logger) *Service {
+	s := &Service{
+		logger:    l,
+		endpoints: c.index(),
+	}
+	return s
+}
+
+type HandlerConfig struct {
+	URL      string `mapstructure:"url"`
+	Endpoint string `mapstructure:"endpoint"`
+}
+
+type Handler struct {
+	s        *Service
+	bp       *bufpool.Pool
+	url      string
+	endpoint string
+	logger   *log.Logger
+}
+
+func (s *Service) Handler(c HandlerConfig, l *log.Logger) alert.Handler {
+	return &Handler{
+		s:        s,
+		bp:       bufpool.New(),
+		url:      c.URL,
+		endpoint: c.Endpoint,
+		logger:   l,
+	}
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+func (s *Service) endpoint(e string) (c Config, ok bool) {
+	s.mu.RLock()
+	c, ok = s.endpoints[e]
+	s.mu.RUnlock()
+	return
+}
+
+func (s *Service) Update(newConfigs []interface{}) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, nc := range newConfigs {
+		if c, ok := nc.(Config); ok {
+			s.endpoints[c.Endpoint] = c
```

Comment:
This does look like it is going to work.

The config updates need to effect live code, meaning if you change the URL for an endpoint then the existing alert handlers and httpPost nodes need to update to the new URL. This should be easily accomplished by locking the `Handler|Endpoint` type NewHTTPRequest method etc.

As it currently stands the configs are read once so these changes never propagate to running code.

Let me know if I can help clarify this more.